app/models/search_vector.rb

```diff
@@ -1,8 +1,76 @@
 class SearchVector < ActiveRecord::Base
+
+  WEIGHT_VALUES = [
+    0.03, # D
+    0.1,  # C
+    0.3,  # B
+    1.0   # A
+  ]
+
+  DISCUSSION_FIELD_WEIGHTS = {
+    'discussions.title'        => :A,
+    'motion_names'             => :B,
+    'discussions.description'  => :C,
+    'motion_descriptions'      => :C,
+    'comment_bodies'           => :D
+  }
+
   belongs_to :discussion
   self.table_name = 'discussion_search_vectors'
 
-  def self.execute_search_query(*args)
-    connection.execute sanitize_sql_array args
+  class << self
+    def index!(discussion_ids)
+      discussion_ids = Array(discussion_ids).map(&:to_i)
+      where(discussion_id: discussion_ids).delete_all
+      discussion_ids.each do |discussion_id|
+        connection.execute index_thread_sql(discussion_id)
+        yield if block_given?
+      end
+    end
+    handle_asynchronously :index!
+
+  end
+
+  def self.index_everything!
+    index_without_delay! Discussion.pluck(:id)
+  end
+
+  def self.index_thread_sql(discussion_id)
+    "INSERT INTO discussion_search_vectors (discussion_id, search_vector)
```

Comment:
This hasn't changed from the previous implementation (although I might be tempted to simplify it some, it works just fine.)

We're taking each of the different parts of the discussion and weighting them based on our weights above, creating a multi-weighted vector which gets stored in SearchVector for searching later.

Note the string_agg functions, which knit together all of the, say, motion names, into a single blob of text for vectorizing.